### ISSUE TITLE
Orientation sensor doen't work properly

### ISSUE THREAD

**Issue by stucki** [aquaris-dev/DEPRECATED_android_device_bq_piccolo#25]
_From @SIRSteiner on December 22, 2015 13:49_

In version from 21.12.2015 the orientation doesn't work properly. After manual reboot it works a while.
Thank you

PS: Bootloader is the new one.

_Copied from original issue: piccolo-dev/android_device_bq_piccolo#25_

**Comment by stucki** [comment]
_From @piwats on December 22, 2015 17:9_

Can confirm. After some time it gets stuck.

**Comment by stucki** [comment]
_From @piwats on December 22, 2015 17:17_

Also the magnetic sensor seems to be affected as well. Compass lags or stays stuck.

**Comment by stucki** [comment]
_From @Kra1o5 on December 23, 2015 12:34_

Can you guys provide a logcat/dmesg?

Thanks

**Comment by stucki** [comment]
_From @SIRSteiner on December 23, 2015 15:50_

[dmesg.zip](https://github.com/piccolo-dev/android_device_bq_piccolo/files/70966/dmesg.zip)
Here is it... 

Kind regards
Ronny Steiner

Am 23. Dezember 2015 13:34:35 MEZ, schrieb Alejandro Lopez notifications@github.com:

**Comment by stucki** [comment]
_From @Kra1o5 on December 23, 2015 16:9_

@SIRSteiner,

Thanks but not help me much.

I think this problem is selinux related.

@eloimuns can you set selinux to permissive to test it?

**Comment by stucki** [comment]
_From @SIRSteiner on December 23, 2015 16:11_

Can I help with other commands on terminal as root? 

Kind regards
Ronny Steiner

**Comment by stucki** [comment]
_From @Kra1o5 on December 23, 2015 16:13_

@SIRSteiner , yes.

But it can be more easy using terminal on PC with:

adb pull sepolicy; adb shell su -c dmesg | audit2allow -p sepolicy

Thanks

**Comment by stucki** [comment]
_From @SIRSteiner on December 23, 2015 16:33_

Here is the first...
Is the second command correct? Pipe in adb shell? I get bash: audit2allow: command not found

Sorry I have to send sepolicy per mail. Upload not possible...
Just a moment...

**Comment by stucki** [comment]
_From @Kra1o5 on December 23, 2015 16:36_

To use audit2allow you need to have it installed on linux pc.

sepolicy file is needed for compare sepolicy & dmesg.

**Comment by stucki** [comment]
_From @SIRSteiner on December 23, 2015 16:45_

I use manjaro. No package found for that. But should I send you the sepolicy and the output of
adb shell su -c dmesg
?

I did it. Look at your mail inbox please

**Comment by stucki** [comment]
_From @Kra1o5 on December 23, 2015 17:10_

You need install: policycoreutils

**Comment by stucki** [comment]
_From @piwats on December 23, 2015 18:5_

From a fresh reinstalled 3.2.2 and then converted to a clean cm-20151221:

audit2allow gives this out:

```
libsepol.policydb_read: policydb magic number 0x5b3e363c does not match expected magic number 0xf97cff8c or 0xf97cff8d
```

**Comment by stucki** [comment]
_From @SIRSteiner on December 23, 2015 18:27_

Thank you piwats, because I am not able to install policycoreutils. It is only available in AUR and there are some errors while compiling dependencies, sorry

**Comment by stucki** [comment]
_From @piwats on December 23, 2015 18:33_

I think my kernel is too new to do a audit2allow. Mine is 4.3.3 vs 3.10.49 on the phone.

**Comment by stucki** [comment]
_From @SIRSteiner on December 23, 2015 18:37_

I sent Kra1o5 both sepolicy and the dmesg output. Maybe he can do
cat dmesg.txt | audit2allow -p sepolicy
with my files?

**Comment by stucki** [comment]
_From @piwats on December 23, 2015 18:42_

I made a video showing the compass being stuck/lagging, if needed I can upload it.

**Comment by stucki** [comment]
_From @Kra1o5 on December 27, 2015 16:20_

Fixed in cm-12.1-20151227-UNOFFICIAL-piccolo.

Greetings
